### PR TITLE
Fix room rank overriding global rank permissions

### PR DESCRIPTION
**Pull request by panpawn**

### CODE REVIEW

File: rooms.js [hunk 1]
```diff
@@ -225,13 +225,17 @@ let Room = (() => {
...
+		let ranks;
 		if (this.auth) {
 			if (user.userid in this.auth) {
-				return this.auth[user.userid];
+				rank = (global && ranks.indexOf(user.group) > ranks.indexOf(this.auth[user.userid]) ? user.group : this.auth[user.userid]);
+				return rank;
 			}
 			if (this.tour && this.tour.room) {
-				return this.tour.room.getAuth(user);
+				rank = (global && ranks.indexOf(user.group) > 0 ? user.group : ' ');
```
Comment (panpawn) [inline rooms.js]: Should I _just_ check for global admin here?

**Comment by Zarel** [comment]
Can you be more specific about the problem this is solving?

**Comment by Zarel** [comment]
(It is generally intentional for room rank to override global rank)

**Comment by panpawn** [comment]
I just setup a vanilla PS server, and I tested a few commands in a battle room (as a player) that didn't work, like /declare and /modnote... this does however deviate from the problems TI is having, but I found that in general, this fixes all of the above issues. (except modlog in battles, but there's a specific check for that in /modlog)

**Comment by Zarel** [comment]
That's intentional.

**Pull request closed without merging**